### ISSUE TITLE
Cannot assign a TypedDict to a superset TypedDict (with Union'ed field)

### ISSUE THREAD

**Issue by peat-psuwit** [microsoft/pyright#9647]
**Describe the bug**
Given the following code:

```python
from typing_extensions import Literal, TypedDict

class Sdp(TypedDict):
  type: Literal["offer", "answer"]

class OfferSdp(TypedDict):
  type: Literal["offer"]

def x(sdp: Sdp):
    pass

def y(offerSdp: OfferSdp, sdp: Sdp):
    x(offerSdp) # (1)
    sdp["type"] = offerSdp["type"] # (2)
```

I get a following error from line (1):

```
Argument of type "OfferSdp" cannot be assigned to parameter "sdp" of type "Sdp" in function "x"
  "type" is an incompatible type
    Type "Literal['offer']" is not assignable to type "Literal['offer', 'answer']"
      "Literal['offer']" is not assignable to type "Literal['answer']
```

But since `Literal['offer', 'answer']` is essentially a `Union` which contain `Literal['offer']`, I'm not sure why Pyright is trying to assign `Literal['offer']` to each value in the union. Especially since direct value assignment (2) doesn't trigger the same error.

**VS Code extension or command-line**

This is code-server v4.96.2 (based on VS Code 1.96.2) with Pyright extension version 1.1.391. Pylance is not used.

**Comment by erictraut** [comment]
Pyright is working correctly here and in accordance with the Python typing spec, so this isn't a bug. For details about TypedDict assignability rules, refer to [this section of the spec](https://typing.readthedocs.io/en/latest/spec/typeddict.html#assignability). 

To avoid the type error, you'd need to mark `type` as `ReadOnly`, like this:

```python
from typing import ReadOnly
from typing_extensions import Literal, TypedDict

class Sdp(TypedDict):
    type: ReadOnly[Literal["offer", "answer"]]

class OfferSdp(TypedDict):
    type: Literal["offer"]

def x(sdp: Sdp):
    pass

def y(offerSdp: OfferSdp, sdp: Sdp):
    x(offerSdp)  # OK
    sdp["type"] = offerSdp["type"]  # Error because "type" is read-only
```

Writable items are invariant, so compatible subclasses cannot redefine them in a covariant manner.
